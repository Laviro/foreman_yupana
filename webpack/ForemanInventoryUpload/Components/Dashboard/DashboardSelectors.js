import { selectForemanInventoryUpload } from '../../ForemanInventoryUploadSelectors';

export const selectDashboard = (state, accountID) =>
  selectForemanInventoryUpload(state).dashboard[accountID] || {};
export const selectUploading = (state, accountID) =>
  selectDashboard(state, accountID).uploading;
export const selectGenerating = (state, accountID) =>
  selectDashboard(state, accountID).generating;
export const selectPollingProcessID = (state, accountID) =>
  selectDashboard(state, accountID).pollingProcessID;
export const selectActiveTab = (state, accountID) =>
  selectDashboard(state, accountID).activeTab;
