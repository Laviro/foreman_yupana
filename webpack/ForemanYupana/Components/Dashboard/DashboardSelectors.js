import { selectForemanYupana } from '../../ForemanYupanaSelectors';

export const selectDashboard = state => selectForemanYupana(state).dashboard;
export const selectBool = state => selectDashboard(state).bool;
