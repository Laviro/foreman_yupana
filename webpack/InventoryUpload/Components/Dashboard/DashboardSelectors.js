import { selectInventoryUpload } from '../../InventoryUploadSelectors';

export const selectDashboard = (state, accountID) =>
  selectInventoryUpload(state).dashboard[accountID] || {};
export const selectUploading = (state, accountID) =>
  selectDashboard(state, accountID).uploading;
export const selectGenerating = (state, accountID) =>
  selectDashboard(state, accountID).generating;
export const selectPollingProcessID = (state, accountID) =>
  selectDashboard(state, accountID).pollingProcessID;
export const selectActiveTab = (state, accountID) =>
  selectDashboard(state, accountID).activeTab;
