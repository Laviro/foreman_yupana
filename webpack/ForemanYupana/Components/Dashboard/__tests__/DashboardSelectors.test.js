import { testSelectorsSnapshotWithFixtures } from 'react-redux-test-utils';
import { selectDashboard, selectBool } from '../DashboardSelectors';

const state = {
  dashboard: {
    bool: false,
  },
};

const fixtures = {
  'should return Dashboard': () => selectDashboard(state),
  'should return Dashboard bool': () => selectBool(state),
};

describe('Dashboard selectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
