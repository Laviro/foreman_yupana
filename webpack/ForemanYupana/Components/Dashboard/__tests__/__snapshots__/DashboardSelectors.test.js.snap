// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard selectors should return Dashboard 1`] = `
Object {
  "bool": false,
}
`;

exports[`Dashboard selectors should return Dashboard bool 1`] = `false`;
